custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator.go

```diff
@@ -97,24 +123,37 @@ func getResourceIDs(list *v1.PodList) []string {
 	return resourceIDs
 }
 
-func (t *Translator) createFilter(podIDs []string, project string, zone string, clusterName string, metricName string) (string, error) {
-	// project_id, cluster_name and pod_id together identify a pod unambiguously
-	projectFilter := fmt.Sprintf("resource.label.project_id = %q", project)
-	zoneFilter := fmt.Sprintf("resource.label.zone = %q", zone)
-	clusterFilter := fmt.Sprintf("resource.label.cluster_name = %q", strings.TrimSpace(clusterName))
-	// container_name is set to empty string for pod metrics
+func joinFilters(filters ...string) string {
+	return strings.Join(filters, " AND ")
+}
+
+func (t *Translator) filterForCluster() string {
+	projectFilter := fmt.Sprintf("resource.label.project_id = %q", t.config.Project)
+	zoneFilter := fmt.Sprintf("resource.label.zone = %q", t.config.Zone)
```

Comment:
What if we run nodes in multiple zones? Could you please add TODO and create a bug entry to figure it out.

Reply:
Does it matter? I think what matters is that (project, zone, cluster name, pod id, metric name) identifies the SD time series unambiguously. The function name may be misleading for the reason you mentioned, what about just renaming/splitting it?

Reply:
Yes. Let's imagine you have pod running a node in zone1, and the adapter runs in zone2. You will query for zone2 metrics while they are attached to zone1.

Reply:
Right, obviously. Added a TODO.